packages/dom/src/dom/is-html-input-element.js

```diff
@@ -0,0 +1,9 @@
+/* eslint-disable jsdoc/valid-types */
+/**
+ * @param {Element | null | undefined} element
```

Comment:
Are we planning to pass `null` or `undefined`? I see in `isNumberInput` we already require an `Element`, so can't we require it here?

Reply:
If this function is used in other places it might be helpful to also handle `null` and `undefined` correctly. On the other hand, might be a YAGNI situation. I'll leave it up to you.

Reply:
It's easy to widen types at any point, but a breaking change to narrow them. If this is ever released accepting `X | null | undefined` , that's more difficult to narrow to `X` than the other way around. A function that accepts `X` can start the wider accepting `X | null | undefined` at any point.

I agree, it might be handy to accept `null` for taking input from things like [`Document.querySelector`](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector) or [`Node.parentNode`](https://developer.mozilla.org/en-US/docs/Web/API/Node/parentNode) that may return `null`. However, I'd avoid widening the type until it's absolutely necessary.